src/Pages/MyCollection/MyCollection.tsx

```diff
@@ -3,31 +3,44 @@ import Phono_Logo from '../../assets/Phono_Logo';
 import SearchBar from '../../components/SearchBar/SearchBar';
 import AlbumInfo from '../../components/AlbumInfo/AlbumInfo';
 import NavBar from '../../components/NavBar/NavBar';
-import { useEffect, useState } from 'react';
+import { useState } from 'react';
 import useMyCollection from '../../utils/useMyCollection';
 import CoverSwiper from '../../components/CoverSwiper/CoverSwiper';
 
 export default function MyCollection(): JSX.Element {
-  const { getMyCollection, collection } = useMyCollection();
+  const [searchQuery, setSearchQuery] = useState('');
+  const { collection, filteredCollection } = useMyCollection(searchQuery);
   const [activeSlide, setActiveSlide] = useState<number>(0);
 
-  useEffect(() => {
-    getMyCollection();
-  }, []);
-
   return (
     <div className={styles.myCollectionPage}>
       <Phono_Logo />
-      <SearchBar placeholder={'Search my Collection'} onSubmit={console.log} />
-      {collection && (
+      <SearchBar
+        placeholder={'Search my Collection'}
+        onSubmit={(search) => setSearchQuery(search)}
+      />
+      {collection && !filteredCollection && (
         <CoverSwiper
           collection={collection}
           changeActiveSlide={(activeSlideIndex) =>
             setActiveSlide(activeSlideIndex)
           }
         />
       )}
-      {collection && <AlbumInfo collection={collection[activeSlide]} />}
+      {filteredCollection && (
+        <CoverSwiper
+          collection={filteredCollection}
+          changeActiveSlide={(activeSlideIndex) =>
+            setActiveSlide(activeSlideIndex)
+          }
+        />
+      )}
+      {collection && !filteredCollection && (
+        <AlbumInfo collection={collection[activeSlide]} />
+      )}
+      {filteredCollection && (
+        <AlbumInfo collection={filteredCollection[activeSlide]} />
+      )}
```

Comment:
this looks a bit convoluted. you could define your content here with an if else statement for either filteredcontent or !filteredcontent

`let content
if(filteredContent) {
content = ...
} else {
 content = ...
}
return {content}
`